blockchain/contracts/ProofOfPhysicalAddress.sol

```diff
@@ -342,11 +342,9 @@ contract ProofOfPhysicalAddress {
         // Remove physical address from list
         uint256 length = users[msg.sender].physicalAddresses.length;
 
-        for (uint256 i = index; i < length - 1; i++) {
-            users[msg.sender].physicalAddresses[i] = users[msg.sender].physicalAddresses[i+1];
+        if (length > 1) {
```

Comment:
Shouldn't this be `if (index != length -1) {` ?

Reply:
Oh, that's much better. Thanks.

Reply:
Done.